libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
```

Comment:
That probably means pushing that logic down to the `v2p_arch` functions, so I propose merging this PR as it stands now and I'll work those changes into another PR.